test/test_parser.cpp

```diff
@@ -5055,17 +5056,21 @@ TEST(Parser_DictionaryObjects)
     bernie_pets.insert("dog1", astro);
     bernie_pets.insert("dog2", snoopy);
 
+    auto charlie_pets = charlie.get_dictionary(col_dict);
+    charlie_pets.insert("dog1", pluto);
+
     adam.set(col_friend, bernie.get_key());
     bernie.set(col_friend, adam.get_key());
 
     auto q = persons->link(col_dict).column<Int>(col_age) > 4;
-    CHECK_EQUAL(q.count(), 1);
+    CHECK_EQUAL(q.count(), 2);
     q = persons->link(col_friend).link(col_dict).column<Int>(col_age) > 4;
     CHECK_EQUAL(q.count(), 1);
 
-    verify_query(test_context, persons, "pets.@values.age > 4", 1);
-    verify_query(test_context, persons, "pets.@values == obj('dog', 'pluto')", 1);
-    verify_query(test_context, persons, "pets.@values == ANY { obj('dog', 'pluto'), obj('dog', 'astro') }", 2);
+    verify_query(test_context, persons, "pets.@values.age > 4", 2);
+    verify_query(test_context, persons, "pets.@values == obj('dog', 'pluto')", 2);
+    verify_query(test_context, persons, "pets.@values != obj('dog', 'pluto')", 2);
+    verify_query(test_context, persons, "pets.@values == ANY { obj('dog', 'lady'), obj('dog', 'astro') }", 2);
```

Comment:
Is this expected to work?   `pets.@values == ANY { obj('dog', 'lady'), obj('dog', 'astro'), NULL }`

Reply:
I suppose this is illegal. Fixed

Reply:
Damn yes. I was tricked by the question.